File: 7/Python/main.py
class Solution(object):
    def reverse(self, x):
        rev = 0
        neg = False
        if x < 0:
            neg = True
            x = -x
        while(x > 0): 
            a = x % 10
            rev = rev * 10 + a 
            x = x / 10
        if neg == True:
            rev = -rev
        if rev < -2**31 or rev > (2**31 - 1):
            return 0
        return rev

File: 7/C++/main.cpp
#include <cmath>

class Solution {
public:
    int reverse(int x) {
        long long rev = 0;
        int a = 0;
        while (x) {
            a = x % 10;
            rev = rev * 10 + a; 
            x = x / 10;
        }
        if (rev < INT_MIN || rev > INT_MAX) {
            return 0;
        }
        else {
            return rev;
        }
    }
};
